### ISSUE TITLE
Facing error while training decision tree model

### ISSUE THREAD

**Issue by lakshay-sharma-2024** [DrugDiscoveryProject/Drug-Discovery#1]
While trying to train a DecisionTreeClassifier, a ValueError is raised during the .fit() step. The error message suggests an unexpected label type was passed to the model. This issue might occur due to a mismatch between the model type and the data format or target values.
Error is in Cell [6] of CircularFingerprint.ipynb .

**Comment by manan3044** [comment]
According to me, the error maybe caused while importing the model from sklearn.

**Comment by madhusudanhasbe** [comment]
@lakshay-sharma-2024 thank you for raising this issue! @manan3044 is pointing towards the issue accurately.

The error is occurring because `DecisionTreeClassifier` is being used with a continuous target variable (`y_train`), which is not supported. Classification models in scikit-learn expect discrete class labels (e.g., 0, 1, 2...), but in this case, it looks like the labels are **floating-point numbers** — indicating a regression task.

If the target is continuous (like predicting a number, not a class), you should use `DecisionTreeRegressor` instead of `DecisionTreeClassifier`. Here's how you can fix the code:
```
from sklearn.tree import DecisionTreeRegressor
dt = DecisionTreeRegressor()
dt.fit(X_train, y_train)
```
Alternatively, if the task is classification but `y_train` contains float-type class labels, use:
```
from sklearn.preprocessing import LabelEncoder

le = LabelEncoder()
y_train_encoded = le.fit_transform(y_train)
dt = DecisionTreeClassifier()
dt.fit(X_train, y_train_encoded)
```

**Comment by lakshay-sharma-2024** [comment]
Thanks for the help! @madhusudanhasbe and @manan3044 your insights and resolution solved my problem.
